Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -2583,19 +2583,22 @@ RefPtr<CSSStyleValue> Parser::parse_number_value(TokenStream<ComponentValue>& to
 
 RefPtr<CSSStyleValue> Parser::parse_number_percentage_value(TokenStream<ComponentValue>& tokens)
 {
-    auto peek_token = tokens.next_token();
-    if (peek_token.is(Token::Type::Number)) {
-        tokens.discard_a_token(); // number
-        return NumberStyleValue::create(peek_token.token().number().value());
-    }
-    if (peek_token.is(Token::Type::Percentage)) {
-        tokens.discard_a_token(); // percentage
-        return PercentageStyleValue::create(Percentage(peek_token.token().percentage()));
-    }
-    if (auto calc = parse_calculated_value(peek_token); calc && calc->resolves_to_number_percentage()) {
-        tokens.discard_a_token(); // calc
-        return calc;
-    }
+    // Parses [<percentage> | <number>] (which is equivalent to [<alpha-value>])
+    if (auto number = parse_number_value(tokens))
+        return number;
+    if (auto percentage = parse_percentage_value(tokens))
+        return percentage;
+
+    return nullptr;
+}
+
```

Comment:
I see why you wanted to shrink this methods, but it was done the previous way to avoid parsing `calc()` multiple times. If we have `calc(10%)` for example, `parse_number_percentage_value()` will now:
1. `parse_number_value()`
	1. Try to parse a number; fail
	2. Try to parse a `calc()`
	3. See that the `calc()` doesn't resolve to a number; fail
2. `parse_percentage_value()`
	1. Try to parse a percentage; fail
	2. Try to parse a `calc()`, again, doing the same work as in step 1.ii
	3. See that the `calc()` resolves to percentage and return it.

Reply:
Indeed, there's an easy optimization here.